src/Core/MessageLogger.cpp

```diff
@@ -17,23 +17,25 @@
 
 #include "Core/MessageLogger.hpp"
 #include "Core/EventLogger.hpp"
+#include "Settings/PreferencesWindow.hpp"
+#include "generated/SettingsHelper.hpp"
 #include <QDateTime>
-#include <QTextBrowser>
-#include <generated/SettingsHelper.hpp>
 
-MessageLogger::MessageLogger(QObject *parent) : QObject(parent)
+MessageLogger::MessageLogger(PreferencesWindow *preferences, QWidget *parent)
+    : QTextBrowser(parent), preferencesWindow(preferences)
 {
-}
-
-void MessageLogger::setContainer(QTextBrowser *container)
-{
-    box = container;
-    LOG_INFO("MessageLogger container has been initialized");
-    box->setOpenExternalLinks(true);
+    connect(this, SIGNAL(anchorClicked(const QUrl &)), this, SLOT(onAnchorClicked(const QUrl &)));
+    setOpenExternalLinks(true);
 }
 
 void MessageLogger::message(const QString &head, const QString &body, const QString &color, bool htmlEscaped)
 {
+    if (body.contains("<a href='#Preferences/"))
+    {
+        LOG_WARN("The message contains \"<a href='#Preferences/\", htmlEscaped is automatically disabled.");
+        htmlEscaped = false;
+    }
```

Comment:
Why not htmlEscape everything except `<a href...>` (anchor) It will be much safer than keeping whole message unescaped. 

So,

```
Please do this here | <a href=..> | or do this.
--------escaped-----| ----html----| --escaped--
```

This way, we can open external links to in browser and other stuff like docs.

Reply:
I don't think it's very easy to parse a `<a/>` element.

Both htmlEscaped and this auto-unescape are designed for compilation errors, where the messages may contain `#include <considered as an HTML tag>` and the messages are sent in signals and logged in MainWindow, etc. 

So we can let Compiler log messages instead of sending signals, explicitly set htmlEscaped to false and remove this auto-unescape.

Reply:
The messages in the MainWindow and the Checker are different, so I decided to use a different signal for messages from the compiler, and only escape the messages from the compiler.